code/modules/station_goals/bsa.dm

```diff
@@ -265,7 +268,12 @@ CREATION_TEST_IGNORE_SUBTYPES(/obj/machinery/power/bsa/full)
 
 /obj/machinery/power/bsa/full/proc/fire(mob/user, turf/bullseye)
 	winding_up = FALSE
-	playsound(get_turf(src), fire_sound, 50, 1, world.maxx)
+	playsound(get_turf(src), fire_sound, 100, 1, world.maxx, pressure_affected = FALSE, ignore_walls = TRUE)
+	for(var/mob/M in GLOB.mob_living_list)
+		if(compare_z(M.get_virtual_z_level(), get_virtual_z_level()))
+			continue
+		shake_camera(M, 15, 1)
```

Comment:
I feel off this. I think you wanted to make mobs' camera shaken in two Zs:
* Where BS was fired from
* Where BS was fired to

If I am right, check this again, but also put `if(M.client)` since shaking a clinetless mob's camera is meaningless.

Reply:
the explosion itself will shake the camera in location where BS was fired too. shake_camera has a check for client
![image](https://github.com/user-attachments/assets/60e486c9-7e41-46da-8004-0df806111011)

Reply:
> the explosion itself will shake the camera in location

Oh, it now makes sense. Would you please put a comment saying that there?
And, I know the proc has a client check, but calling a proc is potentially slow when you need to iterate a lot of mobs there. So, instead of calling a proc then early return, just not calling that proc is faster.
So, sort of nitpick, but doing that so is recommended when you iterate a lot like that.